proposals/index-range-changes.md

```diff
@@ -0,0 +1,297 @@
+# Index and Range Changes
+
+## Summary
+This proposes to make several changes to the 
+[index and range design](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/ranges.md) based on 
+customer feedback: particularly from the CoreFX team and their experiences adding Index / Range support to .NET Core.
+The change is not to the syntax but rather how the langauge maps the syntax to APIs.
+
+## Motivation
+The level of API churn necessary to adopt `Index` and `Range` is quite high today. Every collection type must have at
+least two new indexers added for `Index` and `Range` respectively. The implementation of `Index` is exactly the same 
+for virtually every collection in .NET. The implementation for `Range` is likewise often similar by simply deferring to 
+the underlying collection type.
+
+This work must be done by the collection author. The reliance on indexers as the primary use case means developers can't
+update existing collections using extension methods. This means the adoption of the feature will be slowed by waiting
+for library authors to update their types.
+
+The use of `Index` does come with a small performance hit. Typically a consumer of `Index` will pay an extra branch
+check (`IsFromEnd` and bounds) as they would with a `int` parameter (bounds only). Moving to `Range` doubles this cost
+as it occurs on each `Index`.
+
+This penalty is small and insignificant in the vast majority of code. For highly optimized code though this extra 
+branching can be unacceptable. It simply can't be thought of as a syntactic transformation from a call to 
+`span.Slice(start, end)` to `span[start..end]`. Instead the performance implications must be evaluated on each
+usage. This will likely lead to the feature being banned in certain portions of code bases.
+
+This proposals seeks to remove the abstraction penalties around `Index` and `Range`: both at an API and performance 
+layer. The goal being that most collection types just work today with minimal change and the feature can be adopted
+in existing code without fear of lost performance.
+
+This proposal specifically does not want to change how `Index` and `Range` type binding occurs. The index and range
+expressions continue to have the same syntax and types.
+
+## Detailed Design 
+### Countable Types
+Any type which has an property named `Length` or `Count` with an accessible getter adn a return type of `int` is 
```

Comment:
adn

Reply:
This is how you know it's authentic and I don't have a ghost writer.